### PR TITLE
refactor(sms): optimize lodash import

### PR DESCRIPTION
**Pull request by cbourgois**
### Description of the Change

Bump `lodash` to `^4.17.11` (`pluck` replaced by `map`, [remove `chain` usage](https://github.com/ovh-ux/manager/pull/69/files#diff-63cb1d5aba0b62ba464f54fe348574f1R202) ... )
Optimize lodash import

7d6f294 — refactor(sms): optimize lodash import

/cc @jleveugle @frenautvh @antleblanc

### CODE REVIEW

File: packages/manager/sms/src/sms/compose/telecom-sms-sms-compose.controller.js [hunk 1]
```diff
@@ -402,47 +417,47 @@ export default class {
...
+        }, omit(this.createSms(slotId), 'receivers')).$promise));
       }
     }
 
     return this.$q.all(promises).then((results) => {
-      const totalCreditsRemoved = _.sum(results, 'totalCreditsRemoved');
-      const invalidReceivers = _.chain(results).pluck('invalidReceivers').flatten().value();
-      const validReceivers = _.chain(results).pluck('validReceivers').flatten().value();
+      const totalCreditsRemoved = sumBy(results, 'totalCreditsRemoved');
+      const invalidReceivers = flatten(map(results, 'invalidReceivers'));
```
Comment (cbourgois) [inline packages/manager/sms/src/sms/compose/telecom-sms-sms-compose.controller.js]:
```js
const invalidReceivers = flatten(map(results, 'invalidReceivers'));
```
is equivalent to 

```js
import flatMap from 'lodash/flatMap';

const invalidReceivers = flatMap(results, 'invalidReceivers');
```

cf https://lodash.com/docs/4.17.11#flatMap

File: packages/manager/sms/src/sms/compose/addPhonebookContact/telecom-sms-sms-compose-addPhonebookContact.controller.js [hunk 2]
```diff
@@ -41,16 +53,16 @@ export default class {
       hasBeenAdded: false,
     };
     this.selectedContacts = angular.copy(this.phonebooks.lists);
-    this.availableTypes = _.get(this.constant.SMS_PHONEBOOKS, 'numberFields');
+    this.availableTypes = get(this.constant.SMS_PHONEBOOKS, 'numberFields');
     this.$scope.$watch('AddPhonebookContactCtrl.phonebookContact.sorted', (contacts) => {
-      _.each(contacts, (contact) => {
-        const alreadyAdded = _.find(this.selectedContacts, { id: contact.id });
+      each(contacts, (contact) => {
```
Comment (marie-j) [inline packages/manager/sms/src/sms/compose/addPhonebookContact/telecom-sms-sms-compose-addPhonebookContact.controller.js]: `each` -> `forEach`

File: packages/manager/sms/src/sms/compose/addPhonebookContact/telecom-sms-sms-compose-addPhonebookContact.controller.js [hunk 3]
```diff
@@ -66,39 +78,39 @@ export default class {
...
           });
         }).then((contacts) => {
           const clonedContacts = [];
-          _.each(this.availableTypes, (field) => {
-            _.each(_.cloneDeep(contacts), (contact) => {
-              _.set(contact, 'type', field);
-              _.set(contact, 'id', [contact.id, contact.type].join('_'));
-              if (_.isEmpty(_.get(contact, field))) {
+          each(this.availableTypes, (field) => {
+            each(cloneDeep(contacts), (contact) => {
```
Comment (marie-j) [inline packages/manager/sms/src/sms/compose/addPhonebookContact/telecom-sms-sms-compose-addPhonebookContact.controller.js]: `each` -> `forEach`

File: packages/manager/sms/src/sms/compose/addPhonebookContact/telecom-sms-sms-compose-addPhonebookContact.controller.js [hunk 4]
```diff
@@ -130,8 +142,8 @@ export default class {
       this.phonebookContact.orderBy,
       this.phonebookContact.orderDesc,
     );
-    this.phonebookContact.sorted = _.each(data, (contact) => {
-      _.set(contact, 'isSelected', _.some(this.selectedContacts, { id: contact.id }));
+    this.phonebookContact.sorted = each(data, (contact) => {
```
Comment (marie-j) [inline packages/manager/sms/src/sms/compose/addPhonebookContact/telecom-sms-sms-compose-addPhonebookContact.controller.js]: `each` -> `forEach`

File: packages/manager/sms/src/sms/compose/telecom-sms-sms-compose.controller.js [hunk 5]
```diff
@@ -107,9 +122,9 @@ export default class {
       this.senders.raw = result.senders;
       this.receivers.raw = result.receivers;
       this.phonebooks.raw = result.phonebooks;
-      this.phonebooks.current = _.head(this.phonebooks.raw);
+      this.phonebooks.current = head(this.phonebooks.raw);
       return this.senders.raw;
-    }).then(senders => _.each(senders, (sender) => {
+    }).then(senders => each(senders, (sender) => {
```
Comment (marie-j) [inline packages/manager/sms/src/sms/compose/telecom-sms-sms-compose.controller.js]: `each` -> `forEach`

File: packages/manager/sms/src/sms/compose/telecom-sms-sms-compose.controller.js [hunk 6]
```diff
@@ -378,7 +393,7 @@ export default class {
     });
     modal.result.then((result) => {
       this.phonebooks.lists = [];
-      _.each(result, contact => this.phonebooks.lists.push(contact));
+      each(result, contact => this.phonebooks.lists.push(contact));
```
Comment (marie-j) [inline packages/manager/sms/src/sms/compose/telecom-sms-sms-compose.controller.js]: `each` -> `forEach`

File: packages/manager/sms/src/sms/pending/telecom-sms-sms-pending.controller.js [hunk 7]
```diff
@@ -215,7 +220,7 @@ export default class {
     this.loading.cancelAll = true;
     return this.$q.all([
       this.$timeout(angular.noop, 1000),
-    ].concat(_.each(this.pending.raw, sms => this.api.sms.jobs.delete({
+    ].concat(each(this.pending.raw, sms => this.api.sms.jobs.delete({
```
Comment (marie-j) [inline packages/manager/sms/src/sms/pending/telecom-sms-sms-pending.controller.js]: `each` -> `forEach`

File: packages/manager/sms/src/sms/compose/addPhonebookContact/telecom-sms-sms-compose-addPhonebookContact.controller.js (hunk 2 above)
Comment (FredericEspiau) [reply packages/manager/sms/src/sms/compose/addPhonebookContact/telecom-sms-sms-compose-addPhonebookContact.controller.js]: Interesting, the migration guide doesn't ask to do this: https://github.com/lodash/lodash/wiki/Changelog

It does make sense though

**Pull request merged**